test/e2e_node/node_shutdown_linux_test.go

```diff
@@ -0,0 +1,217 @@
+			f.PodClient().CreateBatch(pods)
+
+			list, err := f.PodClient().List(context.TODO(), metav1.ListOptions{
+				FieldSelector: nodeSelector,
+			})
+			framework.ExpectNoError(err)
+			for _, pod := range list.Items {
+				framework.ExpectEqual(
+					pod.Status.Phase == v1.PodRunning,
+					true,
+					"pod is not ready",
+				)
+			}
+
+			ginkgo.By("Emitting shutdown signal")
+			err = emitSignalPrepareForShutdown(true)
+			framework.ExpectNoError(err)
+
+			// Not critical pod should be shutdown
+			gomega.Eventually(func() error {
+				list, err = f.PodClient().List(context.TODO(), metav1.ListOptions{
+					FieldSelector: nodeSelector,
+				})
+				if err != nil {
+					return err
+				}
+				for _, pod := range list.Items {
+					if kubelettypes.IsCriticalPod(&pod) {
+						if pod.Status.Phase != v1.PodRunning {
+							return fmt.Errorf("critical pod should not be shutdown")
+						}
+					} else {
+						if pod.Status.Phase != v1.PodFailed || pod.Status.Reason != "Shutdown" {
+							return fmt.Errorf("pod should be shutdown")
+						}
+
+					}
+				}
+				return nil
+			}, podStatusUpdateTimeout, pollInterval).Should(gomega.BeNil())
+
+			// All pod should be shutdown
+			gomega.Eventually(func() error {
+				list, err = f.PodClient().List(context.TODO(), metav1.ListOptions{
+					FieldSelector: nodeSelector,
+				})
+				if err != nil {
+					return err
+				}
+				for _, pod := range list.Items {
+					if pod.Status.Phase != v1.PodFailed || pod.Status.Reason != "Shutdown" {
+						return fmt.Errorf("pod should be shutdown")
+					}
+				}
+				return nil
+			},
+				// Critical pod starts shutdown after (nodeShutdownGracePeriod-nodeShutdownGracePeriodCriticalPods)
+				podStatusUpdateTimeout+(nodeShutdownGracePeriod-nodeShutdownGracePeriodCriticalPods),
+				pollInterval).Should(gomega.BeNil())
+		})
+
+		ginkgo.It("should be able to handle a cancelled shutdown", func() {
+			ginkgo.By("Emitting Shutdown signal")
+			err := emitSignalPrepareForShutdown(true)
+			framework.ExpectNoError(err)
+			gomega.Eventually(func() error {
+				isReady := getNodeReadyStatus(f)
+				if isReady {
+					return fmt.Errorf("node did not become shutdown as expected")
+				}
+				return nil
+			}, nodeStatusUpdateTimeout, pollInterval).Should(gomega.BeNil())
+
+			ginkgo.By("Emitting Shutdown false signal; cancelling the shutdown")
+			err = emitSignalPrepareForShutdown(false)
+			framework.ExpectNoError(err)
+			gomega.Eventually(func() error {
+				isReady := getNodeReadyStatus(f)
+				if !isReady {
+					return fmt.Errorf("node did not recover as expected")
+				}
+				return nil
+			}, nodeStatusUpdateTimeout, pollInterval).Should(gomega.BeNil())
+		})
+	})
+})
+
+func getGracePeriodOverrideTestPod(name string, node string, gracePeriod int64, critical bool) *v1.Pod {
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
```

Comment:
I think it is a best practice to namespace pods created for this test with the test namespace